src/renderer/notification.ts

```diff
@@ -0,0 +1,319 @@
+import { ipcMain } from 'electron';
+
+import { createComponentWindow, windowExists } from '../app/window-utils';
+import { AnimationQueue } from '../common/animation-queue';
+import { logger } from '../common/logger';
+import NotificationHandler from './notification-handler';
+
+// const MAX_QUEUE_SIZE = 30;
+const CLEAN_UP_INTERVAL = 60 * 100;
+const animationQueue = new AnimationQueue();
+
+interface ICustomBrowserWindow extends Electron.BrowserWindow {
+    notificationData: INotificationData;
+    displayTimer: NodeJS.Timer;
+    clientId: number;
+}
+
+interface INotificationData {
+    id: number;
+    title: string;
+    text: string;
+    image: string;
+    flash: boolean;
+    color: string;
+    tag: string;
+    sticky: boolean;
+    company: string;
+    displayTime: number;
+}
+
+type startCorner = 'upper-right' | 'upper-left' | 'lower-right' | 'lower-left';
+
+const notificationSettings = {
+    startCorner: 'upper-right' as startCorner,
+    width: 380,
+    height: 100,
+    totalHeight: 0,
+    totalWidth: 0,
+    corner: {
+        x: 0,
+        y: 0,
+    },
+    firstPos: {
+        x: 0,
+        y: 0,
+    },
+    templatePath: '',
+    maxVisibleNotifications: 6,
+    borderRadius: 5,
+    displayTime: 5000,
+    animationSteps: 5,
+    animationStepMs: 5,
+    logging: true,
+};
+
+class Notification extends NotificationHandler {
+
+    private readonly funcHandlers = {
+        onCleanUpInactiveNotification: () => this.cleanUpInactiveNotification(),
+        onCreateNotificationWindow: (data: INotificationData) => this.createNotificationWindow(data),
+    };
+    private readonly activeNotifications: Electron.BrowserWindow[] = [];
+    private readonly inactiveWindows: Electron.BrowserWindow[] = [];
+    private readonly notificationQueue: INotificationData[] = [];
+    private readonly notificationCallbacks: any[] = [];
+    private cleanUpTimer: NodeJS.Timer;
+
+    constructor(opts) {
+        super(opts);
+        ipcMain.on('close-notification', (_event, windowId) => {
+            this.hideNotification(windowId);
```

Comment:
does this destroy the browser window or just hide it?

Reply:
This just hides the browser window

Reply:
But won't that be waste of resources. What happens if we close it?

Reply:
Approx 200mb will be reserved based on the display width for about a min. Without this, every notification will be rendered a fresh which looks delayed

Reply:
Understood, thanks for the explanation.